src/codegen/compile.ml

```diff
@@ -8639,7 +8703,7 @@ module GCRoots = struct
     })
 
   let store_static_roots env =
-    Arr.vanilla_lit env (E.get_static_roots env)
+    Arr.vanilla_lit env Tagged.M (E.get_static_roots env)
```

Comment:
I wonder whether the static roots could be tagged as immutable, as they reside in static space and should not change at runtime (references to static MutBoxes that are not moved by the GC). This would be different with EOP.

Reply:
Even more accurate would be as a tuple, since the contents are not uniformly typed.

Reply:
(But then Motoko tuples cannot contain mutboxs, so this would be cheating too)

Reply:
You are right. `Tagged.M` seems a pragmatic choice.

Reply:
Ok, let's just keep as is